src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -0,0 +1,295 @@
+            /// Writes a sequence of bytes to the current stream.
+            /// </summary>
+            /// <param name="buffer">An array of bytes. This method copies count bytes from buffer to the current stream.</param>
+            /// <param name="offset">The zero-based byte offset in buffer at which to begin copying bytes to the current stream.</param>
+            /// <param name="count">The number of bytes to be written to the current stream.</param>
+            public override void Write(byte[] buffer, int offset, int count)
+            {
+                Span<byte> value = buffer.AsSpan().Slice(offset, count);
+
+                this.StartWritingByteValueInChunks(value);
+            }
+
+            /// <summary>
+            /// Asynchronously writes a portion of a byte array to the underlying stream in chunks.
+            /// </summary>
+            /// <param name="buffer">The byte array from which data will be written.</param>
+            /// <param name="offset">The zero-based byte offset in the buffer at which to begin copying bytes to the stream.</param>
+            /// <param name="count">The maximum number of bytes to write.</param>
+            /// <param name="token">A CancellationToken to observe while waiting for the task to complete.</param>
+            /// <returns>A task representing the asynchronous write operation.</returns>
+            public override async Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken token)
+            {
+                ReadOnlyMemory<byte> value = buffer.AsMemory().Slice(offset, count);
+                await this.StartWritingByteValueInChunksAsync(value);
+            }
+
+            /// <summary>
+            /// Writes the byte value represented by the provided ReadOnlySpan in chunks using Base64 encoding.
+            /// </summary>
+            /// <param name="value">The ReadOnlySpan containing the byte value to be written.</param>
+            private void StartWritingByteValueInChunks(ReadOnlySpan<byte> value)
+            {
+                // Process the input value in chunks
+                for (int i = 0; i < value.Length; i += chunkSize)
+                {
+                    int remainingBytes = Math.Min(chunkSize, value.Length - i);
+                    bool isFinalBlock = false;
+
+                    // Take a chunk of bytes from the input value.
+                    ReadOnlySpan<byte> chunk = value.Slice(i, remainingBytes);
+
+                    // Get unprocessed bytes from the buffer.
+                    ReadOnlySpan<byte> bytesNotProcessedFromPreviousChunk = buffer;
+
+                    // If the buffer is not empty, then we copy the bytes from the buffer
+                    // to the current chunk being processed.
+                    if (!bytesNotProcessedFromPreviousChunk.IsEmpty)
+                    {
+                        byte[] combinedArray = new byte[bytesNotProcessedFromPreviousChunk.Length + chunk.Length];

```

Comment:
Would it be better to rent the byte array from `ArrayPool<byte>`?

Reply:
done